request/src/main/java/com/vimeo/networking2/VimeoApiClient.kt

```diff
@@ -1938,6 +1939,92 @@ interface VimeoApiClient {
+
+    /**
+     * Fetch a [PermissionPolicy].
+     *
+     * @param uri The URI from which content will be requested.
+     * @param fieldFilter The fields that should be returned by the server in the response, null indicates all should be
+     * returned.
+     * @param cacheControl The optional cache behavior for the request, null indicates that the default cache behavior
+     * should be used.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun fetchPermissionPolicy(
+        uri: String,
+        fieldFilter: String?,
+        queryParams: Map<String, String>?,
+        cacheControl: CacheControl?,
+        callback: VimeoCallback<PermissionPolicy>
+    ): VimeoRequest
+
+    /**
+     * Either associates a new [PermissionPolicy] to a [TeamEntity], or replaces the current associated
+     * [PermissionPolicy] with a new one.
+     *
+     * @param permissionPolicyUri Uri from a [PermissionPolicy] you want to newly associate with a [TeamEntity].
+     * @param teamEntityType The type of [TeamEntity] the association is being made with.
+     * @param teamEntityUri The uri of the [TeamEntity], probably from [TeamEntity.uri].
+     * @param uri The URI used to perform the PUT, likely sourced from the edit interaction of [TeamPermission.metadata]
+     * @param callback The callback which will be notified of the request completion.
+     * @param queryParams The query parameters included in the PUT.
+     * @param bodyParams The body parameters included in the PUT.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun putTeamPermission(
+        permissionPolicyUri: String,
+        teamEntityType: TeamEntityType,
+        teamEntityUri: String,
+        uri: String,
+        callback: VimeoCallback<Unit>,
+        queryParams: Map<String, String>? = null,
+        bodyParams: Map<String, Any>? = null
+    ): VimeoRequest
+
+    /**
+     * Removes the associated [PermissionPolicy] from a [TeamEntity].
+     *
+     * @param teamEntityType The type of [TeamEntity] the association is being removed for.
+     * @param teamEntityUri The uri of the [TeamEntity], probably from [TeamEntity.uri].
+     * @param uri The URI used to perform the PUT, likely sourced from the remove interaction of
+     * [TeamPermission.metadata]
+     * @param callback The callback which will be notified of the request completion.
+     * @param queryParams The query parameters included in the DELETE.
+     * @param bodyParams The body parameters included in the DELETE.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun deleteTeamPermission(
+        teamEntityType: TeamEntityType,
+        teamEntityUri: String,
+        uri: String,
+        callback: VimeoCallback<Unit>,
+        queryParams: Map<String, String>? = null,
+        bodyParams: Map<String, Any>? = null
```

Comment:
What `queryParams` and `bodyParams` would we expect a consumer to pass to this function?